src/bindings/c/include/openvino/ov_shape.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for ov_shape C API
+ *
+ * @file ov_shape.h
+ */
+
+#pragma once
+
+#define MAX_DIMENSION 8
+
+/**
+ * @struct ov_shape_t
+ * @brief Reprents a static shape.
+ */
+typedef struct {
+    int64_t rank;
+    int64_t dims[MAX_DIMENSION];
```

Comment:
Honestly... Do we need to have limitation for maximum dimension? Rank should define the size of dimensions array...

Reply:
Static shape defines a fixed max size will be helpful to manage it(extra allocate and free for the array) and user can access it directly without other additional operations. At least, so far there is any case applying exceed 8 dimensions.

Reply:
We can use rank in order to define the fixed size of static shape. The dimensions vector should have the same size as rank.
What if rank is 10 but we cannot add 10 dimensions?

Reply:
1. rank represents the actual dimension size, but it cannot exceed MAX_DIMENSION
2. It is fixed size memory, will be easy to use, such as
    ...
    ov_shape_t shape;
    ...
3. If we don't use such method to pre-allocate space for int64_t dims[MAX_DIMENSION],  we will need explicitly allocate memory for this array, and also release it explicitly.
4. For rank>MAX case, it should be a limitation, however will we see such scenario in short term?

Reply:
So it is C API, if user works with vectors in C, he always should allocate the memory manually. I think it is ok.

Reply:
I'm afraid I cannot agree with you, because your idea will bring lots of inconvenient for user. A simple struct need so many additional work on it.

Reply:
I'm afraid I cannot agree with you, because in your proposal with static array user cannot create any abstract shape.

So man, user uses C API, he should be ready to some C specific constructions. We shouldn't provide C++ API under C view (because it is not possible). If user use C API, he should be ready that he need to manage a memory in some cases.

We can provide some helper like allocate_shape_for_rank, but honesty I against this idea, because as I already said C API should keep C, we shouldn't modify it to C++ (especially if it introduces any limitations in design).

Reply:
done - provide ov_shape_init() and ov_shape_deinit() to manage its internal memory.